stix_shifter_modules/nozomi/stix_translation/query_constructor.py

```diff
@@ -0,0 +1,549 @@
+import re
+import json
+from os import path
+from datetime import datetime, timedelta
+from stix_shifter_utils.utils import logger
+from stix_shifter_utils.stix_translation.src.patterns.pattern_objects import ObservationExpression, \
+    ComparisonExpression, ComparisonComparators, Pattern, CombinedComparisonExpression, CombinedObservationExpression, \
+    ComparisonExpressionOperators
+
+logger = logger.set_logger(__name__)
+
+START_STOP_PATTERN = r"(\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(\.\d+)?Z)"
+STOP_TIME = datetime.utcnow()
+CONFIG_MAP_PATH = "json/config_map.json"
+
+# API query limit is 1350 (MAX_QUERY_LENGTH 1250 +TIMESTAMP_LENGTH 100)
+MAX_QUERY_LENGTH = 1250
+
+
+class FileNotFoundException(Exception):
+    pass
+
+
+class QueryStringPatternTranslator:
```

Comment:
Can you please add additional in-line comments to help explain these methods. Especially justification for some of the API specific things.

For example, the way it has to handle query size is somewhat unique. I'd also like to see the parameters explained a bit better to show what the fields actually are in the method docstring.

Reply:
Updated doc string and additional in-line comments as suggested.